### PR TITLE
DM-11816: Firefly API document update 

### PR DESCRIPTION
**Pull request by cwang2016**
The development
- update markdown file on plotting parameters on 'SurveyKey' and 'SurveyKeyBand'
- update jsdoc mainly by @typedef an object related to plotting parameters defined in WebPlotRequest. 

test:
compile firefly and create jsdoc (gradle :firefly:bAA)
Browse the pages built on on jsdoc from firefly/build/firefly/war/docs/js/

Some issue occurs all over the created pages jsdoc:
open page on global.html, the description line of some members or methods is shown repeatedly.

### CODE REVIEW

File: src/firefly/js/api/ApiHighlevelBuild.js [hunk 1]
```diff
@@ -214,15 +222,86 @@ function buildImagePart(llApi) {
...
     const {RequestType}= llApi.util.image;
 
+
+    /**
+     * @summary object for web plot request
+     * @description Below is a list of predefined parameters available for web plot request.
+     * Some parameters control how to get an image, a image can be retrieved from a service, a url, of a file on the server.
+     * Others control the zoom, stretch, and color, title, and default overlays. There are also parameters to pre-process an
+     * image, such as crop, rotate or flip.
+     * @typedef {object} WebPlotParams
```
Comment (robyww) [inline src/firefly/js/api/ApiHighlevelBuild.js]: I would really like that to be calls a `WebPlotRequest` as well.  But this is a general challenge with have in JavaScript when you want to make a class out of a simple object.  I guess `WebPlotParams` is as good a name as any.

File: src/firefly/js/api/ApiHighlevelBuild.js [hunk 2]
```diff
@@ -260,16 +339,76 @@ function buildImagePart(llApi) {
...
      * @public
-     * @example firefly.showCoverage
+     * @example firefly.showCoverage('myDiv', {gridOn: true})
      */
     const showCoverage= (div,options) => initCoverage(llApi,div,options);
 
+    /**
+     * @summary The plotting function to display a HiPS
+     * @param {String|HTMLDivElement} targetDiv to put the image in.
+     * @param {WebPlotParams} request a request object with Type=='HiPS' used to display a HiPS
```
Comment (robyww) [inline src/firefly/js/api/ApiHighlevelBuild.js]: technically this should be `{WebPlotParams|WebPlotRequest}`

File: src/firefly/js/api/ApiHighlevelBuild.js [hunk 3]
```diff
@@ -260,16 +339,76 @@ function buildImagePart(llApi) {
...
+     * @param {HipsImageConversionSettings} [hipsImageConversion=undefined] if defined, use these parameter to
+     *                                                convert between image and HiPS
+     * @memberof firefly
+     * @public
+     * @example firefly.showHiPS('hipsDIV1',
```
Comment (robyww) [inline src/firefly/js/api/ApiHighlevelBuild.js]: Thanks for making an example.

File: src/firefly/js/api/ApiHighlevelBuild.js [hunk 4]
```diff
@@ -260,16 +339,76 @@ function buildImagePart(llApi) {
...
+
     const showHiPS= (targetDiv, request, hipsImageConversion)  =>
                         showImageInMultiViewer(llApi, targetDiv, request, true, hipsImageConversion);
+
+
+    /**
+     * @summary The plotting function to display a HiPS or an image
+     * @param {String|HTMLDivElement} targetDiv to put the image in.
+     * @param {WebPlotParams} hipsRequest a request object used to display a HiPS
+     * @param {WebPlotParams} imageRequest  a request object used to display an image
```
Comment (robyww) [inline src/firefly/js/api/ApiHighlevelBuild.js]: also `{WebPlotParams|WebPlotRequest}`

File: src/firefly/js/api/ApiHighlevelBuild.js [hunk 5]
```diff
@@ -260,16 +339,76 @@ function buildImagePart(llApi) {
...
+     * @param {number} fovDegFallOver the field of view size to determine when to move between a HiPS and an image
+     * @param {WebPlotParams} allSkyRequest a request object used to display allsky image.
```
Comment (robyww) [inline src/firefly/js/api/ApiHighlevelBuild.js]: also `{WebPlotParams|WebPlotRequest}`

File: src/firefly/js/api/ApiViewer.js [hunk 6]
```diff
@@ -212,6 +212,13 @@ function buildImagePart(channel,file,dispatch) {
         });
     };
 
+    /**
+     * @summary show a HiPS in the firefly viewer in another tab
+     * @param {WebPlotParams} request Web plot request on HiPS type
```
Comment (robyww) [inline src/firefly/js/api/ApiViewer.js]: also `{WebPlotParams|WebPlotRequest}`

File: src/firefly/js/api/ApiHighlevelBuild.js [hunk 7]
```diff
@@ -214,15 +222,86 @@ function buildImagePart(llApi) {
...
+     * @prop {string} Service the service type if Type=='SERVICE', see available services at {@link ServiceType}
+     * @prop {string} plotId plot ID
+     * @prop {string} plotGroupId plot group ID
+     * @prop {String} ObjectName object name that can be looked up by NED or Simbad
+     * @prop {string} Resolver the object name resolver to use, options are: NED, Simbad, NedThenSimbad, SimbadThenNed, PTF.
+     * @prop {string} SizeInDeg the radius or side (in degrees) depending of the service type, used with Type=='SERVICE' or 'HiPS'
+     * @prop {string} SurveyKey the survey, used with  Type='SERVICE'
+     * @prop {string} SurveyKeyBand the survey band, used with Type=='SERVICE' and Service='WISE', 'TWOMASS' or 'ATLAS'
+     * @prop {string} WorldPt target for service request or HiPS request in serialized version
+     * @Prop {string} PlotId plot Id
```
Comment (tgoldina) [inline src/firefly/js/api/ApiHighlevelBuild.js]: Looks like a typo: Prop -> prop.

**Pull request merged**